Cubical/HITs/Nullification/Properties.agda

```diff
@@ -0,0 +1,129 @@
+{-# OPTIONS --cubical --safe #-}
+module Cubical.HITs.Nullification.Properties where
+
+open import Cubical.Foundations.Everything
+open isPathSplitEquiv
+open import Cubical.Modalities.Everything
+open Modality
+open import Cubical.HITs.Localization
+open import Cubical.Data.Unit
+
+open import Cubical.HITs.Nullification.Base
+
+rec : ∀ {ℓ ℓ' ℓ''} {S : Type ℓ} {A : Type ℓ'} {B : Type ℓ''}
+      → (nB : isNull S B) → (A → B) → Null S A → B
+rec nB g ∣ x ∣ = g x
+rec nB g (hub f) = fst (sec nB) (λ s → rec nB g (f s))
+rec nB g (spoke f s i) = snd (sec nB) (λ s → rec nB g (f s)) i s
+rec nB g (≡hub {x} {y} p i) = fst (secCong nB (rec nB g x) (rec nB g y)) (λ i s → rec nB g (p s i)) i
+rec nB g (≡spoke {x} {y} p s i j) = snd (secCong nB (rec nB g x) (rec nB g y)) (λ i s → rec nB g (p s i)) i j s
+
+toPathP⁻ : ∀ {ℓ} (A : I → Type ℓ) {x : A i0} {y : A i1} → x ≡ transport⁻ (λ i → A i) y → PathP A x y
+toPathP⁻ A p i = toPathP {A = λ i → A (~ i)} (sym p) (~ i)
+
+toPathP⁻-sq : ∀ {ℓ} {A : Type ℓ} (x : A) → Square (toPathP⁻ (λ _ → A) (λ _ → transport refl x)) refl
+                                                  (transportRefl x) refl
+toPathP⁻-sq x j i = hcomp (λ l → λ { (i = i0) → transportRefl x j
+                                   ; (i = i1) → x ; (j = i1) → x })
+                          (transportRefl x (i ∨ j))
+
+module _ {ℓ ℓ' ℓ''} {S : Type ℓ} {A : Type ℓ'} {B : Null S A → Type ℓ''} where
+
+  private
+    secCongDep' : ∀ (nB : (x : Null S A) → isNull S (B x)) {x y : Null S A} (p : x ≡ y)
+                  → (∀ (bx : B x) (by : B y) → hasSection (cong₂ (λ x (b : B x) (_ : S) → b) p))
+    secCongDep' nB p = secCongDep (λ _ → const) p (λ a → secCong (nB a))
+
+  ind : (nB : (x : Null S A) → isNull S (B x)) → ((a : A) → B ∣ a ∣) → (x : Null S A) → B x
+  ind nB g ∣ x ∣ = g x
+  ind nB g (hub f)
+    = fst (sec (nB (hub f)))
+          (λ s → transport (λ i → B (spoke f s (~ i))) (ind nB g (f s)))
+
+  ind nB g (spoke f s i)
+    = toPathP⁻ (λ i → B (spoke f s i))
+               (appl ( snd (sec (nB (hub f)))
+                           (λ s → transport (λ i → B (spoke f s (~ i))) (ind nB g (f s))) ) s) i
+
+  ind nB g (≡hub {x} {y} p i)
+    = hcomp (λ k → λ { (i = i0) → transportRefl (ind nB g x) k
+                     ; (i = i1) → transportRefl (ind nB g y) k })
+            (fst (secCongDep' nB (≡hub p) (transport refl (ind nB g x)) (transport refl (ind nB g y)))
+                 (λ i s → transport (λ j → B (≡spoke p s (~ j) i)) (ind nB g (p s i))) i)
+
+  ind nB g (≡spoke {x} {y} p s j i)
+    = hcomp (λ k → λ { (i = i0) → toPathP⁻-sq (ind nB g x) k j
+                     ; (i = i1) → toPathP⁻-sq (ind nB g y) k j
+                     ; (j = i1) → ind nB g (p s i) })
+            (q₂ j i)
+
+    where q₁ : Path (PathP (λ i → B (≡hub p i)) (transport refl (ind nB g x)) (transport refl (ind nB g y)))
```

Comment:
Should these be inlined? Having the types of these explicit was super helpful in actually proving this case, but inlining them might make more apparent its similarities to the other three cases.

Reply:
This probably makes the proof a bit more readable... So let's keep them